roles/workstation/tasks/swap.yml

```diff
@@ -0,0 +1,47 @@
+# get the current swap size and available disk space
+- name: get current swap size
+  ansible.builtin.shell:
+    cmd: swapon --show | grep /swapfile | awk '{print $3}' | sed 's/.$//'
+  register: swapSize
+
+- name: get system RAM in GB
+  set_fact:
+    ram: "{{ ansible_memtotal_mb // 1024 }}" # Convert to GB
+
+- name: get free disk space in GB
+  ansible.builtin.shell:
+    cmd: df -h --block-size=G / | sed 1d | awk '{print $4}' | head -c -2
+  register: diskSpace
+
+- name: set newSwapSize based on diskSpace and RAM
+  set_fact:
+    newSwapSize: "{{ (ram | int * 2) if (diskSpace.stdout | int) > (ram | int * 4) else (ram | int if (diskSpace.stdout | int) > (ram | int * 2) else 2) }}"
```

Comment:
Can you please add a comment here describing the pseudocode intended logic in case we wish to change things in the future.